front/components/assistant/conversation/SidebarMenu.tsx

```diff
@@ -82,6 +86,12 @@ type GroupLabel =
   | "Last 12 Months"
   | "Older";
 
+// Handle "infinite" scroll
+// We only start with 10 conversations shown (no need more on mobile) and load more until we fill the parent container.
+// We use an intersection observer to detect when the bottom of the list is visible and load more conversations.
+// That way, the list starts lightweight and only show more conversations when needed.
+const CONVERSATIONS_PER_PAGE = 10;
```

Comment:
Take it out from the component to not initiate on every re-renders

Reply:
Also we no longer show only 10 since it will be only applied for read conversations but I think it's okay given that unread will be pretty small for now?